lib/ImageMatrix.cpp

```diff
@@ -121,4 +121,71 @@ namespace CIL {
         }
         *this = new_img;
     }
+
+    void ImageMatrix::forEach(
+        std::function<void(const typename ImageMatrix::ValueType&)> fn) const
+    {
+        for (auto i = 0U; i < m_height; ++i)
+        {
+            for (auto j = 0U; j < m_width; ++j)
+            {
+                for (auto k = 0U; k < m_num_components; ++k)
+                {
+                    fn((*this)(i, j, k));
+                }
+            }
+        }
+    }
+
+    void ImageMatrix::forEach(
+        std::function<void(const typename ImageMatrix::ValueType&, uint32_t row,
+                           uint32_t col, uint32_t comp)>
+            fn) const
+    {
+        for (auto i = 0U; i < m_height; ++i)
+        {
+            for (auto j = 0U; j < m_width; ++j)
+            {
+                for (auto k = 0U; k < m_num_components; ++k)
+                {
+                    fn((*this)(i, j, k), i, j, k);
+                }
+            }
+        }
```

Comment:
```suggestion
        for (auto i = 0U; i < m_height; ++i)
            for (auto j = 0U; j < m_width; ++j)
                for (auto k = 0U; k < m_num_components; ++k)
                    fn((*this)(i, j, k), i, j, k);
```
Looks tooo spacy i'll say